lib/resources/forms.js

```diff
@@ -268,6 +268,20 @@ module.exports = (service, endpoint) => {
       .then((form) => Forms.getVersions(form.id, queryOptions))));
 
 
+  ////////////////////////////////////////
+  // RESTORE / UNDELETE
+
+  service.post('/projects/:projectId/forms/:id/restore', endpoint(({ Forms }, { auth, params }) =>
+    Forms.getByProjectAndXmlFormId(params.projectId, params.id, false, null, QueryOptions.none, true)
+      .then(getOrNotFound)
+      .catch((err) => ((err.name === 'DataIntegrityError')
+        ? Promise.reject(Problem.internal.notImplemented({ feature: 'restoring ambiguous xmlFormId' }))
+        : Promise.reject(err)))
+      .then((form) => auth.canOrReject('form.delete', form))
```

Comment:
move this permission check up: fetch it and immediately check auth 

form.delete --> form.restore

Reply:
remove this problem